### DISCUSSION TITLE
Getting apns-unique-id from ios push notification delivery

### DISCUSSION THREAD

**Discussion by galharth** [excid3/noticed#443] [Ideas]
Hello,

Is it possible to get apns-unique-id after the ios push notification is delivered? (it's different than apns-id). 

The apns-unique-id is needed in order to track the notification status on apple cloudKIT

Thanks!

**Comment by excid3** [comment]
We provide the Apnotic object, so you can do anything that Apnotic can do. Please check their docs for more details. 👍

**Comment by galharth** [comment]
@excid3 Thanks for the answer; where can I access the apn response object (Apnotic::Response) after sending the notification? 


I only know how to access the object before the response here:
```
  def ios_format(apn)
      apn.custom_payload = { universal_link: url }
      apn.alert = { title: title(recipient), body: message(recipient) }
      apn.sound = 'default'
      apn.priority = '10'
      apn.expiration = Time.now.to_i + 2.days.to_i
  end
  
```

**Reply by excid3** [reply]
I don't believe there's a callback after sending other than if the delivery failed. What are you trying to do?

Edit: We do have callbacks on delivery methods, but don't think it's accessible unless you inherit from the delivery method.
https://github.com/excid3/noticed/blob/main/lib/noticed/delivery_method.rb#L31-L33

**Reply by galharth** [reply]
I need the callback to get the "apns-unique-id" It's supposed to be inside the apnotic response object header.  I need the apns-unique-id to track the notification status inside Apple cloud kit dashboard. How do I inherit the callback from the delivery method? thanks

**Reply by galharth** [reply]
@excid3 Hi just checking in - I would love an example of how to inherit the callback from the delivery method 🙏
